tools/tsdb-series/main.go

```diff
@@ -136,3 +151,34 @@ func printBlockIndex(ctx context.Context, blockDir string, printChunks bool, mat
 		return
 	}
 }
+
+func formatTime(ts time.Time) string {
+	return ts.UTC().Format(time.RFC3339Nano)
+}
+
+func computeChunkStats(chr tsdb.ChunkReader, chks []chunks.Meta) (time.Time, time.Time, int) {
+	if len(chks) == 0 {
+		return time.Time{}, time.Time{}, 0
+	}
+
+	minTS := chks[0].MinTime
+	maxTS := chks[len(chks)-1].MaxTime
+	totalSamples := 0
+
+	for _, cm := range chks {
+		c, _, err := chr.ChunkOrIterable(cm)
```

Comment:
do you want to panic in case a chunk iterable is returned? from what I see it's possible with native histograms to have multiple chunks for the same `cm` in which case we get an iterator instead of a chunk (`c`)

Reply:
interesting. It shouldn't panic, we check if `c != nil` later, but I can add support for iterable too. From the `ChunkOrIterable` documentation I didn't understand that this was used by NH. I thought it's for OOO samples only, and that's obviously not something we need to deal with when reading blocks.

Reply:
you're right in both cases. I didn't pay enough attention